From 98de7effc934e2aabaa28bc37d7686f988e6461f Mon Sep 17 00:00:00 2001
From: Varun Sharma <vsharma@habana.ai>
Date: Mon, 19 Jul 2021 20:08:31 +0300
Subject: [PATCH] [SW-44448] gaudi2: provide full address of image to lkd

This patch provides full 64 bit address of image addr to lkd during
dynamic load protocol. Preboot uses arm view point of address, so while
passing to lkd, convert to full 64 bit address.

Change-Id: Id445a6881b7557112259ff990000bdea822db4a3
Signed-off-by: Varun Sharma <vsharma@habana.ai>
---

diff --git a/arch/gaudi2/arm/common.c b/arch/gaudi2/arm/common.c
index cdd9f5a..d11439c 100644
--- a/arch/gaudi2/arm/common.c
+++ b/arch/gaudi2/arm/common.c
@@ -83,6 +83,12 @@
 	return boot_strap & strap_mask;
 }
 
+u64 lkd_dev_spec_get_img_addr(uintptr_t base)
+{
+	/* Return 64 bit full address */
+	return ((base ^ SDRAM_BASE) | SDRAM_BASE_FULL);
+}
+
 static void pcie_update_reg_range(u32 addr, u32 addr_end, u32 value)
 {
 	for (; addr <= addr_end ; addr += 4)
diff --git a/arch/gaudi2/arm/common.h b/arch/gaudi2/arm/common.h
index 2d912de..2a1ccb1 100644
--- a/arch/gaudi2/arm/common.h
+++ b/arch/gaudi2/arm/common.h
@@ -11,9 +11,11 @@
 #ifdef TARGET_IS_ONIX
 #define PPBOOT_TEXT_BASE		0x8200000000
 #define SDRAM_BASE			0x8200000000ull
+#define SDRAM_BASE_FULL			0x1001000200000000ull
 #else
 #define PPBOOT_TEXT_BASE		0x7FFD000000
 #define SDRAM_BASE			0x7FFD000000ull
+#define SDRAM_BASE_FULL			0x1000FFFFFD000000ull
 #endif
 
 #define HL_PCIE_LINK_RETRIES		50
